src/lib/APS2.cpp

```diff
@@ -487,58 +528,98 @@ void APS2::write_memory(const uint32_t & addr, const vector<uint32_t> & data){
 	 * data = vector<uint32_t> data
 	 */
 
-	if (data.size() == 0) {
-		return;
-	}
```

Comment:
This empty packet check was important for a certain edge case on the old firmware. Does the unit test suite contain an empty message check?